### DISCUSSION TITLE
schedule thousands of jobs, opinions

### DISCUSSION THREAD

**Discussion by michelson** [sidekiq/sidekiq#6331] [General]
I'm currently developing a message sequencing system where, upon adding contacts to a sequence step, a job is scheduled in Sidekiq. The job data is minimal, consisting only of an ID. My concern is whether this approach will scale efficiently as the system grows.

Has anyone implemented a similar system or have insights on the scalability of scheduling numerous small jobs in Sidekiq? Any advice or best practices would be greatly appreciated.

Thanks in advance!

**Comment by mperham** [comment]
Job scheduling is an O(log N) operation so it should scale reasonably well into thousands, less well into millions. You can always write your own load test to verify.

**Reply by michelson** [reply]
Thanks for the answer, Mike!
